crates/rslint_core/src/groups/errors/no_new_symbol.rs

```diff
@@ -0,0 +1,64 @@
+use crate::rule_prelude::*;
+use ast::*;
+use SyntaxKind::*;
+
+declare_lint! {
+    /**
+    `Symbol` shouldn't be constructed using `new` keyword, instead
+    it should be called as a function.
+
+    ## Incorrect code examples
+
+    ```js
+    // This call results in TypeError
+    const fooSymbol = new Symbol("foo"); 
+    ```
+
+    ## Correct code examples
+
+    ```js
+    const fooSymbol = Symbol("foo");
+    ```
+    */
+    #[derive(Default)]
+    NoNewSymbol,
+    errors,
+    "no-new-symbol",
+}
+
+#[typetag::serde]
+impl CstRule for NoNewSymbol {
+    #[allow(clippy::blocks_in_if_conditions)]
+    fn check_node(&self, node: &SyntaxNode, ctx: &mut RuleCtx) -> Option<()> {
+        if node.kind() == NEW_EXPR
+            && node.to::<NewExpr>().object()?.syntax().text() == "Symbol"
+        {
+            let err = ctx
+                .err(
+                    self.name(),
+                    "`Symbol` cannot be called as a constructor.",
+                )
+                .primary(range, "");
+
```

Comment:
```suggestion
.note(format!("help: call it as a function instead: `{}`", color("Symbol()")))
```
The user should know how to fix the error without looking at its docs